airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -197,17 +197,58 @@ def path(self, **kwargs):
 class InvoiceLineItems(StripeStream):
     """
     API docs: https://stripe.com/docs/api/invoices/invoice_lines
+
+    Line items are part of each 'invoice' record, so use Invoices stream because
+    it allows bulk extraction:
+        0.1.28 and below - 1 request extracts line items for 1 invoice (+ pagination reqs)
+        0.1.29 and above - 1 request extracts line items for 100 invoices (+ pagination reqs)
+
+    if line items object has indication for next pages ('has_more' attr)
+    then use current stream to extract next pages. In major cases pagination requests
+    are not performed because line items are fully reported in 'invoice' record
     """
 
     name = "invoice_line_items"
 
     def path(self, stream_slice: Mapping[str, Any] = None, **kwargs):
         return f"invoices/{stream_slice['invoice_id']}/lines"
 
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+
+        params = super().request_params(stream_state, stream_slice, next_page_token)
+
+        # add 'starting_after' param
+        if not params.get("starting_after") and stream_slice and stream_slice.get("starting_after"):
+            params["starting_after"] = stream_slice["starting_after"]
+
+        return params
```

Comment:
This part of new code is similar for at least 3 streams, we can reuse this code by creating the new major class for these streams and inherit from that, make couple of class properties and change them based on stream we fetch. WDYT?